libyul/AsmJsonConverter.cpp

```diff
@@ -33,23 +33,23 @@ namespace solidity::yul
 
 Json::Value AsmJsonConverter::operator()(Block const& _node) const
 {
```

Comment:
Sounds like something that could be useful to tools. I think it should be properly introduced in a standalone PR though. We need to agree on the name (maybe `"origin"`?) and decide if we want other debug info here, like AST ID. Also needs some tests.

Reply:
I think this should be in the same PR - we are changing the value of "location" in the case of code that has `@use-src`.
origin sounds good.

Reply:
> we are changing the value of "location" in the case of code that has `@use-src`.

I wanted to say that we are not because there are no changes in test output, but you're right, there should have been some changes. So either something's wrong or, more likely, we do not have coverage for this.

Anyway, I'm out of time now so I'll be doing that change tomorrow. I'll do it in the same PR if I manage to before the release but I still think a separate PR would be more convenient - this one is pretty much a bugfix for the fact that these locations changed in 0.8.6 and is basically finished.

Reply:
Ok, I'm fine with either, but it seems that there is a problem in the ast export and re-import anyway...

Reply:
Right. I think that the problem is that the origin location is not preserved in AST nodes in inline assembly when you export and reimport the AST (https://github.com/ethereum/solidity/pull/11998/files#r713769635). So it's exactly what you pointed out but it's not enough to add it to JSON, we also need to import that information back.

I'm going to remove the assertions from this PR and add them in the one that extends the AST instead.

Reply:
Of course, anything you add to the json export also has to be added to the json import :)

Reply:
I knew there was a catch somewhere :P

Anyway, I'm going to actually push one more change. I'll make the importer use the `"src"` location for both native and origin location - I'm not not sure if adding `"origin"` actually makes sense since it has to include file index and I only have one when compiling inline assembly - and there it's the same as `"src"` anyway.

Reply:
> So either something's wrong or, more likely, we do not have coverage for this.

And this turns out to be because without #9590 there's no way to output the Yul AST (other than for inline assembly or generated sources and we can't have debug info annotations there). So naturally it does not get tested.